servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1429,6 +1429,78 @@ Size2 TextureStorage::texture_size_with_proxy(RID p_proxy) {
 	return texture_2d_get_size(p_proxy);
 }
 
+void TextureStorage::texture_rd_initialize(RID p_texture, const RID &p_rd_texture) {
+	ERR_FAIL_COND(!RD::get_singleton()->texture_is_valid(p_rd_texture));
+
+	// TODO : investigate if we can support this, will need to be able to obtain the order and obtain the slice info
+	ERR_FAIL_COND_MSG(!RD::get_singleton()->texture_is_shared(p_rd_texture), "Please create the texture object using the original texture");
+
+	RD::TextureFormat tf = RD::get_singleton()->texture_get_format(p_rd_texture);
+	ERR_FAIL_COND(!(tf.usage_bits & RD::TEXTURE_USAGE_SAMPLING_BIT));
+
+	TextureFromRDFormat imfmt;
+	_texture_format_from_rd(tf.format, imfmt);
+	ERR_FAIL_COND(imfmt.image_format == Image::FORMAT_MAX);
+
+	Texture texture;
+
+	switch (tf.texture_type) {
+		// Q? What to do with _1D and cube textures?
+		case RD::TEXTURE_TYPE_2D: {
+			ERR_FAIL_COND(tf.array_layers != 1);
+			texture.type = TextureStorage::TYPE_2D;
+		} break;
+		case RD::TEXTURE_TYPE_2D_ARRAY: {
+			ERR_FAIL_COND(tf.array_layers == 1);
+			texture.type = TextureStorage::TYPE_LAYERED;
+			texture.layered_type = RS::TEXTURE_LAYERED_2D_ARRAY;
+		} break;
+		case RD::TEXTURE_TYPE_3D: {
+			ERR_FAIL_COND(tf.array_layers != 1);
+			texture.type = TextureStorage::TYPE_3D;
+		} break;
+		default: {
+			ERR_FAIL_MSG("This RD texture can't be used as a render texture");
+		} break;
+	}
+
+	texture.width = tf.width;
+	texture.height = tf.height;
+	texture.depth = tf.depth;
+	texture.layers = tf.array_layers;
+	texture.mipmaps = tf.mipmaps;
+	texture.format = imfmt.image_format;
+	texture.validated_format = texture.format; // ??
+
+	RD::TextureView rd_view;
+	rd_view.format_override = imfmt.rd_format == tf.format ? RD::DATA_FORMAT_MAX : imfmt.rd_format;
```

Comment:
If the user changes the swizzle data when creating their texture, we have no way to obtain that information, we don't store it. So for now I'm assuming the swizzle info was used as we determine it in `_validate_texture_format`.

We may want to enhance the rendering device to record the swizzle information when creating the texture  (slice) so we can retrieve it. Then we can also remove the swizzle data from `_texture_format_from_rd`

Reply:
It is my understanding the `TextureView` overrides but I must be honest that this is an area I haven't fully figured out yet because I have no real world experience with it. Logic suggest that most users will stick to RGBA formats and will never deviate from the required swizzle so it may very well be a non issue in practice.